src/controller/CHIPDeviceController.cpp

```diff
@@ -528,6 +533,52 @@ void DeviceController::OnValue(const char * key, const char * value) {}
 
 void DeviceController::OnStatus(const char * key, Operation op, CHIP_ERROR err) {}
 
+#if CHIP_ENABLE_MDNS
+
+void DeviceController::OnNodeIdResolved(CHIP_ERROR error, uint64_t nodeId, const Mdns::ResolvedNodeData & nodeData)
+{
+    Device * device   = nullptr;
+    bool addressValid = false;
```

Comment:
Under what circumstances would the nodeid resolve an address that is not valid?

Could we have a separate delegate for error vs success? Like 'NodeIdResolved' and 'NodeIdResolutionFailed'?

Reply:
If I'm not mistaken, the current Avahi port may pass an empty address if just SRV record is published and the corresponding A record doesn't exist.